frubesss/heylinda-app
# feat: add cypress e2e tests

## Description

This change adds [cypress](https://www.cypress.io) end to end testing to the web application and adds a GitHub action to run these tests on every PR.

The initial tests added are more smoke end to end style tests just to add the basic foundations. In light of this they can be expanded upon in other PRs.

## Ticket Link

Closes: https://github.com/heylinda/heylinda-app/issues/122

## How has this been tested?

Check actions tab Cypress section for test results. https://github.com/frubesss/heylinda-app/actions/workflows/cypress.yaml

## Test Locally

1. `yarn install`
2. `yarn test:e2e`
3. Run each e2e test via the Cypress GUI

Or headless mode without GUI

1. `yarn install`
2. `yarn test:e2e:headless`
3. Check terminal for test result output

## Checklist

- [x] Added a "Closes [issue number]" in the ticket link section
- [x] UI changes: include screenshots of all affected screens
